src/integration-test/java/com/commercetools/sync/integration/externalsource/cartdiscounts/CartDiscountSyncIT.java

```diff
@@ -308,20 +308,23 @@ void sync_WithUpdatedCartDiscount_WithNewCustomTypeWithWrongResIdentifier_Should
             .join();
 
         //assertions
-        assertThat(errorMessages).containsExactly(
-            "Failed to resolve references on CartDiscountDraft with key:'key_1'. Reason: "
-                + "Failed to resolve custom type reference on CartDiscountDraft with key:'key_1'. Reason: "
-                + "The value of the 'id' field of the Resource Identifier/Reference is blank (null/empty).");
-        assertThat(exceptions)
-            .hasSize(1)
-            .hasOnlyOneElementSatisfying(exception -> {
-                assertThat(exception).isInstanceOf(ReferenceResolutionException.class);
+        assertThat(exceptions).hasSize(1);
+        assertThat(errorMessages).hasSize(1);
```

Comment:
a test assertion should not expect it with if, you need to expect it directly, please add additional tests for different cases.

Reply:
The if-else block has been removed in latest change.